### DISCUSSION TITLE
How to execute jobs immediately based on their priority?

### DISCUSSION THREAD

**Discussion by amitonlentra** [kagkarlsson/db-scheduler#484] [Q&A]
We have a requirement to execute jobs based on its priority (custom priority like high, medium, low). The jobs are **not** to be scheduled for later execution but to be executed based on priority. 

Can someone guide us with sample code on how could this library be used for such requirement?

Thanks.

**Comment by amitonlentra** [comment]
@kagkarlsson - your inputs on this would be really helpful to us in understanding how can we use this library for our use case. Thank you in advance.

**Reply by kagkarlsson** [reply]
Currently, the earliest execution-time has the highest priority, since polling orders executions by this field. So a hack might be to schedule high-priority jobs to yesterday or something like that. Not bulletproof (e.g. if volumes are very high and queue is never really empty), but maybe helpful in your case?

**Reply by amitonlentra** [reply]
> Currently, the earliest execution-time has the highest priority, since polling orders executions by this field. So a hack might be to schedule high-priority jobs to yesterday or something like that

Sounds like a good workaround. We have 3 predefined priorities - high, medium and low. Based on the above details, high jobs can be scheduled for yesterday, medium jobs can be scheduled for the current time and low priority jobs can be scheduled for current time + 10 mins. I am assuming based on this logic, if we have 50 medium and low priority jobs currently submitted/executing and if we submit a high priority job, the library would pick up the high priority job for execution as soon as one thread from the pool becomes free.

> Not bulletproof (e.g. if volumes are very high and queue is never really empty)

I didn't catch this scenario that won't work with the above workaround. Could you please elaborate?
